core/src/main/java/org/radargun/SlaveBase.java

```diff
@@ -73,6 +74,17 @@ protected void scenarioLoop() throws IOException {
             } else if (!stage.shouldExecute()) {
                log.info("Stage '" + stage.getName() + "' should not be executed");
                response = new DistStageAck(state);
+               if (stage instanceof AbstractDistStage) {
```

Comment:
The issue is that `AbstractDistStage.shouldExecute()` is called before `AbstractDistStage.getExecutingSlaves`. In the case that the stage specifies a non-existing group, `AbstractDistStage.shouldExecute()` returns `false`, so `AbstractDistStage.getExecutingSlaves` isn't executed and neither the benchmark or stage fails.